src/sql/workbench/contrib/notebook/test/browser/outputProcessor.test.ts

```diff
@@ -0,0 +1,166 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as assert from 'assert';
+import { nb } from 'azdata';
+import * as op from 'sql/workbench/contrib/notebook/browser/models/outputProcessor';
+import { JSONObject } from 'sql/workbench/services/notebook/common/jsonext';
+import { nbformat as nbformat } from 'sql/workbench/services/notebook/common/nbformat';
+import { getRandomElement } from 'vs/base/common/arrays';
+import { getRandomString } from 'vs/editor/test/common/model/linesTextBuffer/textBufferAutoTestUtils';
+
+
+suite('OutputProcessor functions', function (): void {
+	const text = getRandomString(1, 10);
+
+	const arbitraryMetadata = {
+		// arbitrarily construction chart options. It is any JSONObject
+		azdata_chartOptions: {
+			scale: true,
+			dataGrid: false,
+			arbitraryCharProperty: {
+				prop1: 'value1',
+				prop2: {
+					deepProp1: 3
+				}
+			}
+		}
+	};
+
+	suite('getData', async function (): Promise<void> {
+		// data tests
+		for (const outputType of ['execute_result', 'display_data', 'update_display_data'] as nbformat.OutputType[]) {
+			const output = <nbformat.IExecuteResult>{
+				output_type: outputType,
+				data: <op.IMimeBundle>{
+					result: text  // free form fields of the IMimeBundle. This is just an arbitrarily cooked-up hardcoded example for the test.
+				}
+			};
+			test(`test for outputType:${output.output_type}`, async () => {
+				verifyGetDataForDataOutput(output);
+			});
+		}
+
+		// error tests
+		for (const traceback of [undefined, [undefined], [], [''], [getRandomString(0, 10), getRandomString(0, 10)]]) {
```

Comment:
Arguably your test shouldn't have a random test case - the getRandomString() could cause some tests to pass and some tests to fail.

Reply:
Fuzz tests can be useful, but yeah I agree for these it'd be better to keep most tests deterministic and then optionally we can add fuzz tests if that seems useful.

Reply:
'could' is the key word. The claim with these tests is that they should not fail for any of the values. If they do it needs to be investigated. This is symbolizing any payload given to the input should get returns as its. I disagree with hardcode value in these test scenarios even though they could ostensibly lead to more deterministic test.

Reply:
Just because I have chosen a random string does not make it a fuzz test. I do very much think that a random payload is a correct thing.

Reply:
Yeah I suppose it's fine. My concern is that this could fail on someone else's PR and they're going to have to waste time realizing the test failure isn't even related to their work.

I agree with the concept of fuzzy testing but I've usually seen it in cases where the unit test is going to run a few thousand iterations of itself to catch issues.

Fuzzy testing when running one test per build seems like it'd just create noise and not get the attention of the correct person to look into any potential failure.

Reply:
I agree with Amir. These tests should focus on testing a set of known values to verify behavior. If you want to make a special fuzz test that's fine (and useful) but it should be separate and clearly indicated that it's using random values so someone looking at failures can understand why a particular value was "expected".

Reply:
I will change to a known string for know. So should testing with random inputs be separate run?

Reply:
fixed.

Reply:
I think just having a test that specifically runs through some number of random strings is fine - just make sure the title clearly indicates it as such. That'll at least prevent people from ratholing too hard on why the test is failing in most cases. 

What Amir said about the test failing for other people randomly is a slight concern - but since we don't currently have a good system set up for grouping up "types" of tests I don't think it's worth trying to come up with that now. If these start to be an issue for some reason then we can re-evaluate but I don't expect them to actually fail, and if they do fixing it should be simple I would imagine.

Reply:
Looking over the product code the fields in questions are just passed back as output in a different shape of any object and there is no processing done on the contents of the string field in question. So for these apps at least I do not see ROI in writing separate tests for running with multiple random invites. Even the execution times for those tests would seem an unnecessary expense. I would vote on skip authoring of separate tests with random inputs.